File: 1400/books/main.cpp
#include <iostream>

using namespace std;

inline void fastIO()
{
    ios::sync_with_stdio(false);
    cin.tie(NULL);
    cout.tie(NULL);
}

int main()
{
    fastIO();
    int n, t;
    cin >> n >> t;
    int v[n+1];
    v[0] = 0;
    int max_count = 0;
    for (int i=1, pos = 0, sum = 0, count = 0; i<=n; i++)
    {
        cin >> v[i];
        sum += v[i];
        count++;
        while (sum > t)
        {
            sum -= v[++pos];
            count--;
        }
        if (count > max_count) max_count = count;
    }
    cout << max_count << '\n';
}

File: 1400/books/main.py
import sys

def main():
    n, t = [int(x) for x in input().split()]
    v = (n+1) * [0]
    max_count = 0
    pos = 0
    sum = 0
    count = 0
    i = 1
    while i <= n:
        v[i] = readInt()
        sum += v[i]
        count += 1
        while sum > t:
            pos += 1
            sum -= v[pos]
            count -= 1
        if count > max_count:
            max_count = count
        i += 1
    print(max_count)

def readInt():
    s = str()
    c = sys.stdin.read(1)
    while c != ' ' and c != '\n':
        s += c
        c = sys.stdin.read(1)
    return int(s)

if __name__ == '__main__':
    main()
